File: Leet 3546/solution.cs
public class Solution
{
    public bool CanPartitionGrid(int[][] grid)
    {
        int rows = grid.Length;
        int cols = grid[0].Length;

        long total = 0;
        for(int i = 0; i < rows; i++)
        {
            for(int j = 0; j < cols; j++)
            {
                total += grid[i][j];
            }
        }

        long currentSum = 0;
        long hTotal = total;
        for(int i = 0; i < rows; i++)
        {
            for(int j = 0; j < cols; j++)
            {
                currentSum += grid[i][j];
                hTotal -= grid[i][j];
            }

            if (currentSum == hTotal)
            {
                return true;
            }
        }

        currentSum = 0;
        long vTotal = total;
        for(int j = 0; j < cols; j++)
        {
            for(int i = 0; i < rows; i++)
            {
                currentSum += grid[i][j];
                vTotal -= grid[i][j];
            }
            
            if (currentSum == vTotal)
            {
                return true;
            }
        }

        return false;
    }
}

File: Leet 3546/solution.py
class Solution:
    def canPartitionGrid(self, grid: list[list[int]]) -> bool:
        rows: int = len(grid)
        cols: int = len(grid[0])
        total: int = 0
        for i in range(rows):
            for j in range(cols):
                total += grid[i][j]

        current_sum: int = 0
        h_total: int = total
        for i in range(rows):
            for j in range(cols):
                current_sum += grid[i][j]
                h_total -= grid[i][j]
            if current_sum == h_total:
                return True
        
        current_sum = 0
        v_total: int = total
        for j in range(cols):
            for i in range(rows):
                current_sum += grid[i][j]
                v_total -= grid[i][j]
            if current_sum == v_total:
                return True

        return False


def main() -> None:
    print('3546. Equal Sum Grid Partition I')
    sol = Solution()
    grid: list[list[int]] = [[1,4],[2,3]]
    test_case1: str = 'pass' if sol.canPartitionGrid(grid) else 'fail'
    print(f'Case 1 {grid} should equal true? {test_case1}')


if __name__ == '__main__':
    main()